### PR TITLE
Digi_10_createAddOn

### PR DESCRIPTION
**Pull request by bommavikas**
1)Added AddOnController To Create AddOn
2)Added AddOnDataModel

### CODE REVIEW

File: DigiOrderApi/DigiOrderApi/Controllers/AddOnController.cs [hunk 1]
```diff
@@ -0,0 +1,45 @@
...
+        /// </summary>
+        /// <returns></returns>
+        [Authorize(Roles = "ENTERPRISE")]
+        [Route("api/AddOn/Create")]
+        [HttpPost]
+        public HttpResponseMessage createAddOn(JObject data)
+        {
+            try
+            {
+
```
Comment (anebhargavkrishna) [inline DigiOrderApi/DigiOrderApi/Controllers/AddOnController.cs:33]: Remove line 33

Comment (bommavikas) [reply DigiOrderApi/DigiOrderApi/Controllers/AddOnController.cs]: Removed Line 33

File: DigiOrderApi/DigiOrderApi/Controllers/AddOnController.cs [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+    {
+        AddOnDataModel addonDataModel = null;
+        DataTable dtRecords = null;
+
+        // POST: api//// <summary>
+        /// Author:Vikas Bomma
+        /// Date: 7/14/2020
+        /// Description:Post An AddOn into DataBase
+        /// Ref. StoryId: DIGI 10
+        /// Reviewer:
```
Comment (anebhargavkrishna) [inline DigiOrderApi/DigiOrderApi/Controllers/AddOnController.cs:22]: Add reviewer name

File: DigiOrderApi/DigiOrderApi/Controllers/AddOnController.cs [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+        {
+            try
+            {  
+                addonDataModel = new AddOnDataModel();
+                dtRecords = addonDataModel.postAddOn(data);
+                dtRecords.TableName = "ADDONS";
+            }
+            catch (Exception ex)
+            {
+                return Request.CreateResponse(HttpStatusCode.NotFound, "Record Not Inserted");
```
Comment (anebhargavkrishna) [inline DigiOrderApi/DigiOrderApi/Controllers/AddOnController.cs:39]: cahnge error message to internal server error

File: DigiOrderApi/DigiOrderApiDataAccess/AddOnDataModel.cs [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+{
+    public class AddOnDataModel
+    {
+        DataTable dtResult = null;
+
+        //Author:Vikas Bomma
+        //7-1-2020
+        //Description: Post an addon
+        //Reviewer
+        //Review Version:1.0
```
Comment (anebhargavkrishna) [inline DigiOrderApi/DigiOrderApiDataAccess/AddOnDataModel.cs:16]: Remove commetns

File: DigiOrderApi/DigiOrderApi/Controllers/AddOnController.cs (hunk 2 above)
Comment (bommavikas) [reply DigiOrderApi/DigiOrderApi/Controllers/AddOnController.cs]: Added

File: DigiOrderApi/DigiOrderApi/Controllers/AddOnController.cs (hunk 3 above)
Comment (bommavikas) [reply DigiOrderApi/DigiOrderApi/Controllers/AddOnController.cs]: Changed

File: DigiOrderApi/DigiOrderApiDataAccess/AddOnDataModel.cs (hunk 4 above)
Comment (bommavikas) [reply DigiOrderApi/DigiOrderApiDataAccess/AddOnDataModel.cs]: Removed

File: DigiOrderApi/DigiOrderApi/Controllers/TaxpageController.cs [hunk 5]
```diff
@@ -0,0 +1,51 @@
...
+                {
+                    Content = new StringContent(string.Format("Error - formate")),
+                    ReasonPhrase = "Please Enter proper data"
+                };
+                throw new HttpResponseException(response);
+                //return Request.CreateErrorResponse(HttpStatusCode.BadRequest, "Please Enter proper data");
+            }
+            return Request.CreateResponse(HttpStatusCode.OK, dtRecords);
+        }
+    }
```
Comment (anebhargavkrishna) [inline DigiOrderApi/DigiOrderApi/Controllers/TaxpageController.cs:50]: Revert this file

File: DigiOrderApi/DigiOrderApiDataAccess/TaxpageDataModel.cs [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+                dtResult = DB_UTILITY.RunSP("USP_INSERTTAXPAGE", ArrParameter);
+            }
+            catch (Exception ex)
+            {
+                throw ex;
+            }
+            return dtResult;
+        }
+    }
+}
```
Comment (anebhargavkrishna) [inline DigiOrderApi/DigiOrderApiDataAccess/TaxpageDataModel.cs:34]: revert this file

File: DigiOrderApi/DigiOrderApiDataAccess/AddOnDataModel.cs [hunk 7]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
```
Comment (anebhargavkrishna) [inline DigiOrderApi/DigiOrderApiDataAccess/AddOnDataModel.cs:1]: add csproj files

File: DigiOrderApi/DigiOrderApi/Controllers/AddOnController.cs [hunk 8]
```diff
@@ -0,0 +1,44 @@
+﻿using DigiOrderApiDataAccess;
```
Comment (anebhargavkrishna) [inline DigiOrderApi/DigiOrderApi/Controllers/AddOnController.cs:1]: add .csproj file

Comment (bommavikas) [reply DigiOrderApi/DigiOrderApi/Controllers/AddOnController.cs:1]: Added

File: DigiOrderApi/DigiOrderApiDataAccess/AddOnDataModel.cs (hunk 7 above)
Comment (bommavikas) [reply DigiOrderApi/DigiOrderApiDataAccess/AddOnDataModel.cs:1]: Added

File: DigiOrderApi/DigiOrderApi/Controllers/TaxpageController.cs (hunk 5 above)
Comment (bommavikas) [reply DigiOrderApi/DigiOrderApi/Controllers/TaxpageController.cs]: Reverted

File: DigiOrderApi/DigiOrderApiDataAccess/TaxpageDataModel.cs (hunk 6 above)
Comment (bommavikas) [reply DigiOrderApi/DigiOrderApiDataAccess/TaxpageDataModel.cs]: reverted

File: DigiOrderApi/DigiOrderApi/DigiOrderApi.csproj [hunk 9]
```diff
@@ -103,6 +103,7 @@
   </ItemGroup>
   <ItemGroup>
     <Compile Include="App_Start\WebApiConfig.cs" />
+    <Compile Include="Controllers\AddOnController.cs" />
     <Compile Include="Controllers\Appliescontroller.cs" />
     <Compile Include="Controllers\CategoryController.cs" />
     <Compile Include="Controllers\MenuItemController.cs" />
```
Comment (anebhargavkrishna) [inline DigiOrderApi/DigiOrderApi/DigiOrderApi.csproj:109]: Revert this file

File: DigiOrderApi/DigiOrderApiDataAccess/DigiOrderApiDataAccess.csproj [hunk 10]
```diff
@@ -41,6 +41,7 @@
     <Reference Include="System.Xml" />
   </ItemGroup>
   <ItemGroup>
+    <Compile Include="AddOnDataModel.cs" />
     <Compile Include="APIUserTokenDataModel.cs" />
     <Compile Include="AppliesDataModel.cs" />
     <Compile Include="CategoryDataModel.cs" />
```
Comment (anebhargavkrishna) [inline DigiOrderApi/DigiOrderApiDataAccess/DigiOrderApiDataAccess.csproj:47]: Revert this file

**Pull request merged**